CMakeLists.txt

```diff
@@ -18,6 +18,13 @@ project(pybind11)
 set(PYBIND11_MASTER_PROJECT OFF)
 if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
   set(PYBIND11_MASTER_PROJECT ON)
+  set(PYBIND11_EXPORT_NAME "${PROJECT_NAME}Targets")
+  set(PYBIND11_INSTALL_EXPORT ON)
```

Comment:
It's not really a user-facing option, but a project level option; it should not be in the cache.